roles/kubernetes/control-plane/templates/kubeadm-controlplane.v1beta2.yaml.j2

```diff
@@ -19,3 +19,10 @@ controlPlane:
 nodeRegistration:
   name: {{ kube_override_hostname|default(inventory_hostname) }}
   criSocket: {{ cri_socket }}
+{% if inventory_hostname in groups['kube_control_plane'] and inventory_hostname not in groups['kube_node'] %}
+  taints:
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
```

Comment:
there are two taints: `node-role.kubernetes.io/master` and `node-role.kubernetes.io/control-plane` this should account for both.

Reply:
If we take a look at `roles/kubernetes/control-plane/templates/kubeadm-config.v1beta2.yaml.j2` we see that we have the following taint configured:

```
{% if inventory_hostname in groups['kube_control_plane'] and inventory_hostname not in groups['kube_node'] %}
  taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
{% else %}
  taints: []
```

That is because Kubernetes have not replaced taint `master` with `control-plane` yet.

I oppened [this PR ](https://github.com/kubernetes-sigs/kubespray/pull/8044) to do this refactor, but as @oomichi were answered in https://github.com/kubernetes/website/pull/29935#issuecomment-934636434, the migration won't be available until Kubernetes v1.24.

Also if we put both taints, both tolerations will be needed to allow scheduling in `control-plane` nodes. Altough most of de `kubernetes-apps`  deployed with Kubespray support both tolerations, we shouldn't make this mandatory since until Kubernetes v1.24 `node-role.kubernetes.io/control-plane` taint won't be the standard.

Reply:
Thanks for the explanation @unai-ttxu!

/ok-to-test